distributednetworkmanager/src/main/java/com/gruppo1/distributednetworkmanager/ActionsBuilder.java

```diff
@@ -0,0 +1,102 @@
+ * @param <R> a resource type
+ * @param <N> a Node of the network
+ *
+ * Each response is based on a request
+ * Responses (ANSW) are built starting from the request (mainly to retrieve the action ID/number)
+ */
+public interface ActionsBuilder<T extends NodeActionStructure, P extends Peer, R extends Resource, N extends Node> {
+
+    /**
+     * @param peer the peer that will be pinged
+     * @return an action built to execute the ping command to the peer
+     */
+    T buildPing(P peer);
+    /**
+     * @param request the ping request received from a peer (that will be the receiver for the response)
+     * @return a response action built to respond to the received ping command from the peer
+     */
+    T buildPingAnsw(T request);
+
+    /**
+     * @param peer the receiving peer (the peer to invite)
+     * @return an action built to execute the invite command to the peer
+     */
+    T buildInvite(P peer);
+    /**
+     * @param request the received invite request
+     * @param accepted set to indicate if the answer accepts the invite or if it rejects it
+     * @return a response action built to respond to the received invite
+     */
+    T buildInviteAnsw(T request, boolean accepted);
+
+
+    /**
+     * @param peer the receiving peer
+     * @param node the node to be stored
+     * @return an action built to execute the store (node) command to the peer
+     */
+    T buildStore(P peer, N node);
+    /**
+     * @param peer the receiving peer
+     * @param resource the resource to be stored
+     * @return an action built to execute the store (resource) command to the peer
+     */
+    T buildStore(P peer, R resource);
+    /**
+     * @param request the received store (node) request action
+     * @param peers the peers of which forward addresses
+     * @return an array of responses (one for each peer in peers) for the received request
+     */
+    T[] buildStoreAnsw(T request, P[] peers);
+    /**
+     * @param request the received store (resource) request action
+     * @param confirmStore set to confirm that the store command has been executed
+     * @return a response action built to respond to the received store (resource)
+     */
+    T buildStoreAnsw(T request, boolean confirmStore);
+
+
+    /**
+     * @param peer the peer that will receive the built action command
+     * @param node the node of which find its address (on the used network)
+     * @return an action built to execute the find node command to the peer
+     */
+    T buildFindNode(P peer, N node);
+
+    /**
+     * @param request the received find node request from a peer
+     * @param peers the peers that are closest to the searched node (or the searched node's corresponding peer)
+     * @return an array of responses (one for each peer in peers) for the received request
+     */
+    T[] buildFindNodeAnsw(T request, P[] peers);
+
+
+    /**
+     * @param peer he peer that will receive the built action command
```

Comment:
I think here is the , and same here I don't understand why is it necessary to have the parameter peer